### ISSUE TITLE
String type

### ISSUE THREAD

**Issue by aghoneim92** [mun-lang/mun#418]
I would like to add the `String` type to mun. Any pointers on where to start would be greatly appreciated :)

**Comment by Wodann** [comment]
It might be good for us to discuss a design during our next Mun jam and provide that as a guideline for how it can be implemented.

In general, I'd be in favour of implementing it in the Mun standard library. The less complicated our compiler and runtime, the better. However, we'll have to take several things into account that make it hard to say one way or another - at this point:

- ability to marshal a string - using the C ABI - in both `mun_runtime` and our C++ Runtime
- ability to hot reload strings

**Comment by catchouli** [comment]
Ah bummer, I wanted to use this in my next project after seeing the 0.4.0 release on reddit, but after integrating it into my application (which was amazingly simple) and trying to write a logging function I quickly realized that strings aren't supported at all yet!

Shame too, this seems like an amazing project otherwise and seems plenty mature for early use, and I'm looking forward to using it in the future when strings are supported. 👍

**Comment by Wodann** [comment]
> Is there still no string support?

No. We currently have higher priority tasks on our roadmap, so it's unlikely that strings will land in the near future.

**Comment by trsh** [comment]
> > Is there still no string support?
> 
> No. We currently have higher priority tasks on our roadmap, so it's unlikely that strings will land in the near future.

I am currently storing it into a u8 array, but its a bit of struggle. Is the roadmap public?

**Comment by Wodann** [comment]
We discussed the design of strings in Mun. We've settled on:

- UTF-8 encoding
- Implement the `String` type and methods using compiler intrinsics that reuse Rust functionality
